pkg/instance/pool.go

```diff
@@ -0,0 +1,277 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package instance
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/vm"
+)
+
+type Instance interface {
+	Index() int
```

Comment:
Is Index used? I see you duplicate it in poolInstance.idx and use that.
If not, this can be `io.Closer`.

Reply:
It's used by the implementations of `Runner[T Instance]` -- they may need to know some ID of the instance. Though if we get rid of that in `syz-manager` and just use some random names, we can probably live without it.

Reply:
I thought about `io.Closer()`, but then we'd need to refactor `vm` to make `Close()` return `error`. Is that fine?

Reply:
> It's used by the implementations of Runner[T Instance] -- they may need to know some ID of the instance.

That's all generics, so if the concrete type has Index, runner can use it, right?

> I thought about io.Closer(), but then we'd need to refactor vm to make Close() return error. Is that fine?

Yes, using standard signature is good.

Reply:
Dropped `Index()` from the interface and switched `vm` to `io.Closer`.